certbot-dns-rfc2136/certbot_dns_rfc2136/_internal/dns_rfc2136.py

```diff
@@ -186,37 +180,121 @@ def del_txt_record(self, record_name: str, record_content: str) -> None:
             raise errors.PluginError('Received response from server: {0}'
                                      .format(dns.rcode.to_text(rcode)))
 
-    def _find_domain(self, record_name: str) -> str:
+    def _find_domain(self, record_name: str) :
```

Comment:
While I do support this PR, I wouldn't be surprised if there's type errors when you merge in `main`. This PR seems to introduce least a couple of places where the type annotations don't match the docstring, and `mypy` is [run in strict mode now](https://github.com/certbot/certbot/commit/0dedef801ac3499552fba4252ed96c53879799b2).